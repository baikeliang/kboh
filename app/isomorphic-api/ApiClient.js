import 'babel-polyfill'
import Promise from 'bluebird'
import fetch from 'isomorphic-fetch'
import 'url-search-params-polyfill'

console.log('ssaVVVVVVVVVVV!!!!!!!')
if(typeof window !=="undefined")
console.log(window.URLSearchParams)

var URLSearchParams = (typeof window !=="undefined")?window.URLSearchParams:require('urlsearchparams').URLSearchParams;
//import { URLSearchParams } from 'urlsearchparams'

const methods = ['GET', 'POST'];
export default class ApiClient {
    constructor(args) {
        // code
        methods.forEach((method) =>
            this[method] = (path, { params, data } = {} ,{format,done,error}) => {
                ///////////////////////////////
                var u = new URLSearchParams();
                console.log(method)
                var req = {
                    method: method,
                    mode: 'cors',
                    cache: 'default'
                }
                if ((method == 'GET') && params) {
                    for (var key of Object.keys(params)) {
                      u.append(key, params[key])
                    }
                } else if (method == 'POST' && data) {

                    req.body = JSON.stringify(data)
                    req.headers = {
                        'Content-Type': 'application/json'
                    }

                }
                console.log('8888888888'+path+u)
                return fetch(path + u, req).then(format).then(done,error)
            }
        )
    }

    // methods
}
