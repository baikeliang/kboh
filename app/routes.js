import React from 'react';


import { Router, DefaultRoute, Route, NotFoundRoute, Redirect } from 'react-router';

import ToOrder from 'app/toorder/main.js'

import UserCenter from 'app/usercenter/main.js'

import MyOrders from 'app/usercenter/myorder/myorders.js'
import MyOrderInfo from 'app/usercenter/myorder/myorderinfo.js'
import MyCases from 'app/usercenter/mycase/mycases.js'
import MyCaseDesc from 'app/usercenter/mycase/mycasedesc.js'
import Login from 'app/usercenter/login.js'


const App = ({ children }) => (
  <div>
    {children || "ddddd"} 
  </div>
)

function requireAuth(nextState, replaceState) {
  /*
  if ((__SERVER__==false)&&localStorage&&(!localStorage.token))
      replaceState({ nextPathname: nextState.location.pathname }, '/login')
  */
  
  console.log("require.....auth")
  
}

const routes = (
      <Route path="/" component={ App }>
        <Route path="usercenter(?)" component={ UserCenter } onEnter={requireAuth}>
          <Route path="toOrder"   component={ ToOrder }/>
          <Route path="myOrders"  component={ MyOrders } />
          <Route path="myOrderInfo" component={ MyOrderInfo } />
          <Route path="myCases"   component={ MyCases }/>
          <Route path="myCaseDesc"    component={ MyCaseDesc }/>
        </Route>        
        <Route path="login" component={ Login } />  
      </Route>
)

export default routes

