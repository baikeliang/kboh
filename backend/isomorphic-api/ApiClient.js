import 'babel-polyfill'
import Promise from 'bluebird'
//import fetch from 'isomorphic-fetch'
import 'isomorphic-fetch'
import 'url-search-params-polyfill'

var URLSearchParams = (typeof window !=="undefined")?window.URLSearchParams:require('urlsearchparams').URLSearchParams;
//import { URLSearchParams } from 'urlsearchparams'

const methods = ['GET', 'POST'];
export default class ApiClient {
    constructor(args) {
        // code
        methods.forEach((method) =>
            this[method] = (path, { params, data } = {} ,{format,done,error}) => {
                ///////////////////////////////
                var u = new URLSearchParams();
                var req = {
                    method: method,
                    mode: 'cors',
                    cache: 'default'
                }
                if ((method == 'GET') && params) {
                    for (var key of Object.keys(params)) {
                      u.append(key, params[key])
                    }
                } else if (method == 'POST' && data) {
                    console.log(data)
                    console.log("pppppppp")
                    req.body = JSON.stringify(data)
                    console.log(req.body)
                    req.headers = {

                    };

                }
                return fetch(path + u, req).then(format).then(done,error)
            }
        )
    }

    // methods
}
