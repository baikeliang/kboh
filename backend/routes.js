import React from 'react';


import { Router, DefaultRoute, Route, NotFoundRoute, Redirect,IndexRoute} from 'react-router';


import DashBoard  from 'backend/useradmin/main.js'

import Login from 'backend/login.js'


const App = ({ children }) => (
  <div>
    {children || "ddddd"}
  </div>
)

const UserAdmin =  ({ children }) => (
  <div>
    {children}
  </div>
)

function requireAuth(nextState, replaceState) {

  //if ((__SERVER__==false)&&localStorage&&(!localStorage.token))
  //    replaceState({ nextPathname: nextState.location.pathname }, '/login')


  console.log("require.....auth")

}

const routes = (
      <Route path="/" component={ App }>
        <Route path="useradmin(?)" component={ UserAdmin } onEnter={requireAuth}>
          <IndexRoute component={ DashBoard }/>
        </Route>
        <Route path="login" component={ Login } />
      </Route>
)

export default routes

