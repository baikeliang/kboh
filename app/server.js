/*eslint-disable no-console */
import express from 'express'
import serialize from 'serialize-javascript'

import webpack from 'webpack'
import webpackDevMiddleware from 'webpack-dev-middleware'
import webpackConfig from '../webpack.client.config.js'

import React from 'react'
import { renderToString } from 'react-dom/server'
import { Provider } from 'react-redux'
import { createMemoryHistory, match, RouterContext } from 'react-router'
import { syncHistoryWithStore } from 'react-router-redux'

import { configureStore } from './configure-store.js'

const app = express()

app.use(webpackDevMiddleware(webpack(webpackConfig), {
  publicPath: webpackConfig.output.publicPath,
  stats: {
    colors: true
  },
  resolve: {
        alias: {
            app: webpackConfig.resolve.alias.app
        }
    }
}))

import routes from './routes'

const HTML = ({ content, store }) => (
  <html>
    <body>
      <div id="root" dangerouslySetInnerHTML={{ __html: content }}/>
      <div id="devtools"/>
      <script dangerouslySetInnerHTML={{ __html: `window.__initialState__=${serialize(store.getState())};` }}/>
      <script src="/bundle.js"/>
    </body>
  </html>
)

app.use(function (req, res) {
  const memoryHistory = createMemoryHistory(req.url)
  const store = configureStore(memoryHistory)
  const history = syncHistoryWithStore(memoryHistory, store)

  match({ history, routes, location: req.url }, (error, redirectLocation, renderProps) => {
    if (error) {
      res.status(500).send(error.message)
    } else if (redirectLocation) {
      res.redirect(302, redirectLocation.pathname + redirectLocation.search)
    } else if (renderProps) {
      const content = renderToString(
        <Provider store={store}>
          <RouterContext {...renderProps}/>
        </Provider>
      )

      res.send('<!doctype html>\n' + renderToString(<HTML content={content} store={store}/>))
    }
  })
})

app.listen(3000, function () {
  console.log('Server listening on http://localhost:3000, Ctrl+C to stop')
})

