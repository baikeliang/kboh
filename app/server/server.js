/*eslint-disable no-console */
import express from 'express'
import serialize from 'serialize-javascript'
import path from 'path'
import compression from 'compression'
import webpack from 'webpack'
import webpackDevMiddleware from 'webpack-dev-middleware'
import webpackConfig from '../../webpack.client.config.js'

import Immutable from 'immutable'

import React from 'react'
import { renderToString } from 'react-dom/server'
import { Provider } from 'react-redux'
import { createMemoryHistory, match, RouterContext } from 'react-router'
import { syncHistoryWithStore } from 'react-router-redux'

import { ReduxAsyncConnect, loadOnServer } from 'redux-connect'

import { configureStore } from 'app/configure-store.js'

import routes from 'app/routes'

import jwt from 'jsonwebtoken'

import ApiClient from 'app/isomorphic-api/ApiClient'

//import { URLSearchParams } from 'url-search-params-polyfill'

import { URLSearchParams } from 'urlsearchparams'

import Promise from 'bluebird'

var cookieParser = require('cookie-parser')

//var URLSearchParams = URLSearchParams || require('../urlsearchparams').URLSearchParams;

global.__CLIENT__ = false;
global.__SERVER__ = true;


const app = express()

app.use(compression())

app.use(cookieParser())

app.use(express.static(path.join(__dirname, '..', 'build')));


const HTML = ({ content, store }) => (
  <html lang="en">
    <head>
        <meta charSet="utf-8"/>
        <title>登陆</title>
        <meta name="viewport" content="width=device-width, initial-scale=1, minimum-scale=1, maximum-scale=1"/>
        <link rel="stylesheet" href="/app.css"/>
    </head>
    <body>
      <div id="root" dangerouslySetInnerHTML={{ __html: content }}/>
      <div id="devtools"/>
      <script dangerouslySetInnerHTML={{ __html: `window.__initialState__=${serialize(store.getState())};` }}/>
      <script src="/vendor.js"/>
      <script src="/bundle.js"/>
      <script src="/require.js"/>
    </body>
  </html>
)

app.get('/wxcallback/auth',function(req,res){  //wx interface type to decide which to route

    console.log('enter /wxcallback/auth !!!!!')
    /*
    var code = req.query.code
    var type = req.query.type
    */
    var code = 'haha'



    if (code === undefined || code === null) {


    } else {
        /// sync get openid
        let openid = 'asdaaaaa'

        let u = new URLSearchParams();

        console.log('KKKKKKKK'+u)

        u.append('openid', openid);
        res.redirect('/usercenter?'+u)//just sample use type to switch

    }


});

app.use(function (req, res, next) {
  const initialState = Immutable.Map();
  const memoryHistory = createMemoryHistory(req.url)
  const client = new ApiClient(req)
  const store = configureStore(memoryHistory,client,initialState)
  const history = syncHistoryWithStore(memoryHistory, store,{
  selectLocationState (state) {
    return state.get('routing').toJS();
  }
})

  if(req.url.indexOf('/wxcallback')==0){ //wx just jump
      next();
      return;
  }

  match({ history, routes, location: req.url }, (error, redirectLocation, renderProps) => {
    if (error) {
      res.status(500).send(error.message)
    } else if (redirectLocation) {
      res.redirect(302, redirectLocation.pathname + redirectLocation.search)
    } else if (renderProps) {
      loadOnServer({ ...renderProps, store, params:{ req }}).then(() => {
      // 2. use `ReduxAsyncConnect` instead of `RoutingContext` and pass it `renderProps`
      const content = renderToString(
        <Provider store={store} key="provider">
          <ReduxAsyncConnect {...renderProps} />
        </Provider>
      )
      //console.log("~~~~~~~~"+store.getState().auth.token)
      //res.setHeader("Set-Cookie", ['tokenbohe='+store.getState().auth.token]);
      res.send('<!doctype html>\n' + renderToString(<HTML content={content} store={store}/>))

    },(err)=>{

    })
    }
  })
})

app.listen(3000, function () {
  console.log('Server listening on http://localhost:3000, Ctrl+C to stop')
})


