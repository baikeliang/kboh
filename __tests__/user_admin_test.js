import Promise from 'bluebird';
import React from 'react';
import {
  Provider,
  connect
} from 'react-redux';
import { Router,
  createMemoryHistory,
  match,
  Route,
  IndexRoute
} from 'react-router';

import {
  createStore,
  combineReducers
} from 'redux';
import { combineReducers as combineImmutableReducers } from 'redux-immutable';
import {
  mount,
  render
} from 'enzyme';
import { spy } from 'sinon';
import { default as Immutable } from 'immutable';

import routes from 'backend/routes.js'

import {
  configureStore,
  DevTools
} from 'backend/configure-store'

import ApiClient from 'backend/isomorphic-api/ApiClient'

import {
  auth_success,
  auth_is_not_valid,
  auth_response_error_500,
  auth_success_serverrender_openid,
  auth_failed_serverrender_openid,
  resetMock as resetMockAuth
} from '../__mocks__/backend/server_auth_mocker.js'

import {
  load_users_success
} from '../__mocks__/backend/server_user_patient_mocker.js'

import {
  endGlobalLoad,
  beginGlobalLoad
} from '../node_modules/redux-connect/lib/store';
import AsyncConnect from '../node_modules/redux-connect/lib/components/AsyncConnect';

import {
  mountToJson,
  renderToJson
} from 'enzyme-to-json';

import { loadOnServer } from 'redux-connect'

import { renderToString } from 'react-dom/server'

import UserAdmin  from 'backend/index.js'

import { load as loadUsers } from 'backend/redux/reducers/user_patient.js'

describe('behavior', function suite() {

  const endGlobalLoadSpy = spy(endGlobalLoad);
  const beginGlobalLoadSpy = spy(beginGlobalLoad);

  const ReduxAsyncConnect = connect(null, {
    beginGlobalLoad: beginGlobalLoadSpy,
    endGlobalLoad: endGlobalLoadSpy,
  })(AsyncConnect);

  pit('auth success to index', function test(){

    window.__SERVER__ = false;

    var preloadstate = Immutable.fromJS({})

    var client = new ApiClient()

    const history = createMemoryHistory();

    var store = configureStore(history, client, preloadstate)

    const proto = ReduxAsyncConnect.WrappedComponent.prototype;



    spy(proto, 'loadAsyncData');
    spy(proto, 'componentDidMount');

    var reloadOnPropsChange = (props, nextProps) => {
        // reload only when path/route has changed
        return props.location.pathname !== nextProps.location.pathname;
    };
    var location = history.createLocation('/useradmin');

    history.push('/useradmin');

    auth_success();

    var wrapper = mount(
                  <Provider store={store} key="provider">
                      <Router routes={routes} location={location} render={(props) => {
                          return (<ReduxAsyncConnect {...props} helpers={{client}} filter={item => !item.deferred} reloadOnPropsChange={reloadOnPropsChange}/>)}
                      }  history={history}>
                      </Router>
                  </Provider>
                 );


    expect(proto.loadAsyncData.calledOnce).toBe(true);
    expect(proto.componentDidMount.calledOnce).toBe(true);



    expect(beginGlobalLoadSpy.called).toBe(true);
    beginGlobalLoadSpy.reset();

    return proto.loadAsyncData.returnValues[0].then(() => {
      expect(endGlobalLoadSpy.called).toBe(true);
      var state = store.getState().toJS();
      expect(!!state.auth.user).toBe(true)
      expect(mountToJson(wrapper.find('.container-left'))).toMatchSnapshot();
      endGlobalLoadSpy.reset();
      proto.loadAsyncData.restore();
      proto.componentDidMount.restore();
      resetMockAuth();
    })

  });



})
