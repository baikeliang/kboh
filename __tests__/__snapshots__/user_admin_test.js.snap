exports[`behavior auth success to index 1`] = `
<div
  className="container-left">
  <h3 />
  <ul
    className="list">
    <li>
      <label
        className=""
        onClick={[Function]}>
        <a
          className="Appointment"
          href="#">
          <b>
            预约表
          </b>
        </a>
      </label>
    </li>
    <li>
      <label
        className=""
        onClick={[Function]}>
        <a
          className="User"
          href="#">
          <b>
            客户管理
          </b>
        </a>
      </label>
    </li>
    <li>
      <label
        className=""
        onClick={[Function]}>
        <a
          className="DoctorAdmin"
          href="#">
          <b>
            医生管理
          </b>
        </a>
        <i
          className="" />
      </label>
    </li>
    <li>
      <label
        className=""
        onClick={[Function]}>
        <a
          className="Clinicmain"
          href="#">
          <b>
            诊所管理
          </b>
        </a>
        <i
          className="" />
      </label>
    </li>
    <li>
      <label
        className=""
        onClick={[Function]}>
        <a
          className="Record"
          href="#">
          <b>
            病例库
          </b>
        </a>
      </label>
    </li>
    <li>
      <label
        className=""
        onClick={[Function]}>
        <a
          className="Company"
          href="#">
          <b>
            合作企业
          </b>
        </a>
      </label>
    </li>
    <li>
      <label
        className=""
        onClick={[Function]}>
        <a
          className="Bill"
          href="#">
          <b>
            账单管理
          </b>
        </a>
      </label>
    </li>
    <li>
      <label
        className=""
        onClick={[Function]}>
        <a
          className="System"
          href="#">
          <b>
            系统管理
          </b>
        </a>
        <i
          className="" />
      </label>
    </li>
    <li>
      <label
        className=""
        onClick={[Function]}>
        <a
          className="Priceform"
          href="#">
          <b>
            价格表
          </b>
        </a>
        <i
          className="" />
      </label>
    </li>
  </ul>
</div>
`;
